src/pip/_internal/operations/install/wheel.py

```diff
@@ -639,6 +641,14 @@ def _generate_file(path, **kwargs):
         installer_file.write(b'pip\n')
     generated.append(installer_path)
 
+    # Document source package hash
+    hash_file_path = os.path.join(dest_info_dir, 'HASH')
+    with open(hash_file_path, 'w') as dest_hash_file:
+        for alg in STRONG_HASHES:
+            dest_hash_file.write('{}:{}\n'.format(
+                alg, hash_of_file(cast('str', wheel_zip.filename), alg)))
```

Comment:
Why the `cast` to `str` in `wheel_zip.filename`? Can we not pass the filename directly to `hash_of_file` ?

Also the `cast` might not work because `filename` is of type [Text](https://github.com/python/typeshed/blob/master/stdlib/2and3/zipfile.pyi#L159)

Reply:
The type is `unicode`, so I cast to `str`. I can try to change the type of `hash_of_file`, but it seems like casting is easier